samples: nrf9160: modem_shell: remove device reboot from modem DFU (#10995)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

* :ref:`modem_shell_application` sample:

  * Added:

>     * Added sending of GNSS data to carrier library when the library is enabled.

Reviewer: ```suggestion
    * Sending of GNSS data to carrier library when the library is enabled.
```

  * Updated:

>     * Updated the sample to use defines from the :ref:`lib_nrf_cloud` library for string values related to nRF Cloud.

Reviewer: ```suggestion
    * The sample now uses defines from the :ref:`lib_nrf_cloud` library for string values related to nRF Cloud.
```

      Removed the inclusion of the file :file:`nrf_cloud_codec.h`.

>     * Updated modem FOTA to update the firmware without rebooting the application.

Reviewer: ```suggestion
    * Modem FOTA now updates the firmware without rebooting the application.
```
